### PR TITLE
Provide SQLCipher passphrase as Data

### PR DESCRIPTION
**Pull request by eburns-vmware**
Added the ability to specify passphrase for database encryption as a Data object.
Also added resetBytes call for Data objects created when using a string passphrase.
Updated the documentation with a section about passphrase datatypes.
Unit tests.

### Pull Request Checklist

- [x] This pull request is submitted against the `development` branch.
- [x] Inline documentation has been updated.
- [x] README.md or another dedicated guide has been updated.
- [x] Changes are tested.

### CODE REVIEW

File: GRDB/Core/Database.swift
Comment (groue) [inline GRDB/Core/Database.swift:1240]: I could check that previous GRDB versions would also throw a DatabaseError of code SQLITE_ERROR (1). So this check does not introduce any breaking change 👍

File: GRDB/Core/Database.swift
Comment (groue) [inline GRDB/Core/Database.swift:1241]: Please don't quote the method name in the error message. This leads to copy-paste errors (see `changePassphrase` below). And this is not the common practice in this repo. `invalid or empty passphrase` is enough.

File: README.md
Comment (groue) [inline README.md:6441]: Your paragraph has no logical connection to the previous paragraphs:

- Managing the lifetime of the passphrase string (high-level)
- Managing the lifetime of the passphrase bytes (as low-level as it can be)

Your contribution, in order to be understood, and useful, must enrich the story. This is how it will help *everybody*, not just you.

Did you notice the "Managing the lifetime of the passphrase bytes" paragraph? It already contains everything you already need. For your contribution to be useful, you must add something.

File: README.md
Comment (groue) [inline README.md:6447]: This somewhat duplicates what is already written in the "Managing the lifetime of the passphrase bytes" paragraph.

File: README.md
Comment (groue) [inline README.md:6451]: Ha, **THIS** is your real contribution 👍 😄 

That's what you should focus on!

File: GRDB/Core/Database.swift
Comment (groue) [reply GRDB/Core/Database.swift:1240]: Actually, why do check for empty strings, since you check for empty data? And why is the check for empty data necessary?

**Review by groue — CHANGES_REQUESTED** [review]
Thank you, @eburns-vmware!

This is a good pull request, although I'll appreciate if you perform a few changes, and enhance the documentation. I know pretty well that documentation is hard, but it's worth it. I hope I provided sufficient guidance. Please ask if you feel stuck.

File: GRDB/Core/Database.swift
Comment (eburns-vmware) [reply GRDB/Core/Database.swift:1240]: I was frustrated that the SQLite Error 1 was happening and breaking an existing test.  But I agree...this doesn't add value and I'll remove it.

**Comment by groue** [comment]
@eburns-vmware, I saw your changes, but I'm not sure you want me to perform a second review. Please tell me when you think you're ready!

**Comment by eburns-vmware** [comment]
I think it is ready – go ahead and review.

Thanks!

**Comment by groue** [comment]
Thank you very much @eburns-vmware, I like it very much!

Would you accept an invitation that grants you a push access to GRDB? Feel no obligation: this is an open source project, fed on our free time, our needs, our pleasure, and I plan on keeping the lead on it. Yet it is good for everyone that people of various levels of Swift and SQLite expertise can act on the repository when it becomes necessary, so you'll be very welcome!

**Comment by groue** [comment]
Shipped in v5.3.0

**Comment by eburns-vmware** [comment]
Sure, I would accept an invitation.  Thanks for the help getting my contribution into shape.  This is my first open source contribution, and hopefully not my last.

**Comment by groue** [comment]
You're invited, and welcome, and thank you for your first contribution 👏

**Pull request merged**
